src/components/HURUmap/Map/index.js

```diff
@@ -0,0 +1,175 @@
+import { makeStyles } from "@material-ui/core/styles";
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+import { MapContainer, ZoomControl, TileLayer, Pane } from "react-leaflet";
+
+import Layers from "./Layers";
+
+import "leaflet/dist/leaflet.css";
+
+const useStyles = makeStyles(({ palette, typography }) => ({
+  map: {
+    "& .tooltipPop": {
+      background: palette.background.default,
+      boxShadow: "0px 3px 6px #00000029",
+      height: typography.pxToRem(36),
+      width: typography.pxToRem(88),
+      "& .level": {
+        fontWeight: "bold",
+        color: palette.text.secondary,
+        background: palette.primary.main,
+        textTransform: "upperCase",
+        fontSize: typography.pxToRem(7),
+        marginTop: typography.pxToRem(-15),
+        borderRadius: typography.pxToRem(4),
+        paddingTop: typography.pxToRem(2),
+        width: typography.pxToRem(62),
+        height: typography.pxToRem(17),
+        margin: "0 auto",
+        display: "flex",
+        justifyContent: "center",
+      },
+      "& .name": {
+        textAlign: "center",
+        fontSize: typography.pxToRem(9),
+        fontWeight: "bold",
+        textTransform: "capitalize",
+        marginTop: typography.pxToRem(5),
+      },
+    },
+  },
+}));
+
+const preferredChildrenObj = {
+  country: ["county"],
+};
+
+function Map({
+  center,
+  zoom,
+  styles,
+  geometries,
+  geography,
+  tileLayers,
+  ...props
+}) {
+  const classes = useStyles(props);
+  const [selectedBoundary, setSelectedBoundary] = useState(null);
+
+  const getSelectedBoundary = (level, geoms) => {
+    const preferredChildren = preferredChildrenObj[level];
+    if (!preferredChildren) return null;
+
+    const availableLevels = preferredChildren.filter((l) => geoms.children[l]);
+
+    if (availableLevels.length > 0) {
+      const preferredLevel = availableLevels[0];
+      return geoms.children[preferredLevel];
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    let selectedBound =
+      getSelectedBoundary(geography.level, geometries) ?? geometries.boundary;
+
+    if (selectedBound?.type === "Feature") {
+      selectedBound = {
+        ...selectedBound,
+        properties: {
+          ...selectedBound.properties,
+          selected: true,
+        },
+      };
+    } else {
+      // else its a featurecollection
```

Comment:
```suggestion
      // else its a feature collection
```
Or is it `FeatureCollection`?

Reply:
yeah, its a one word. FeatureCollection